Lib/collections/__init__.py

```diff
@@ -973,10 +973,17 @@ def __delitem__(self, key): del self.data[key]
     def __iter__(self):
         return iter(self.data)
 
-    # Modify __contains__ to work correctly when __missing__ is present
+    # Modify __contains__ and get() to work like dict
+    # does when __missing__ is present.
     def __contains__(self, key):
         return key in self.data
 
+    def get(self, key, default=None):
+        if key in self:
```

Comment:
This will have a race condition. Is this not a concern?

Reply:
It's not threadsafe, but it's not a concern. Most of the stdlib is not threadsafe and as the rest of the Python types, have not thread safety guarantees.

Reply:
What do you think about:
```python
    def get(self, key, default=None):
        return self.data.get(key, default)
```

Reply:
What happens if someone overrides `__contains`__?

Reply:
I'm not sure I get your point, could you show an example where this method would cause issues?

Reply:
You'll break this class:

    class AlwaysReturnA(UserDict):
        def __getitem__(self, key):
            return "A"

Reply:
> You'll break this class:

I would say that this class is broken anyway:

```
>>> from collections import UserDict
>>> class AlwaysReturnA(UserDict):
...     def __getitem__(self, key):
...         return "A"
... 
>>> d = AlwaysReturnA()
>>> list(d.keys())
[]
>>> list(d.values())
[]
```

I'm not sure it would make it more broken.

Reply:
Yeah, but you get my idea. Lots of code, even one that I myself wrote uses the fact `get()` calls `__getitem__()`.
It was never promised though, and isn't part of dict's implementation either, so I guess with can break it...? Maybe?